### PR TITLE
parameter sharing in fluid with simple test cases

### PR DESCRIPTION
**Pull request by yu239-zz**
Let me know if some parameter sharing scenarios are needed by not supported by the current infra.

### CODE REVIEW

File: layers/layer_wrappers.py [hunk 1]
```diff
@@ -0,0 +1,523 @@
...
+
+        def __call__(self,
+                     input,
+                     num_flatten_dims=1,
+                     param_attr=None,
+                     bias_attr=None,
+                     use_mkldnn=False,
+                     act=None,
+                     is_test=False,
+                     name=None):
```
Comment (emailweixu) [inline layers/layer_wrappers.py:136]: I feel most arguments except "input" and "is_test" should be set at \_\_init\_\_()

**Review by emailweixu — COMMENTED** [review]
Let's put all python code under python/pprl/

File: layers/tests/test_param_sharing.py [hunk 2]
```diff
@@ -0,0 +1,70 @@
...
+
+
+class TestParamSharing(unittest.TestCase):
+    def __init__(self, *args, **kwargs):
+        super(TestParamSharing, self).__init__(*args, **kwargs)
+        self.fc1 = layers.fc(64)
+        self.fc2 = layers.fc(64)
+        self.embedding = layers.embedding((100, 64))
+        ## we bind the paras of self.embedding to those of self.fc1
+        layers.paras_binding(src_layer=self.fc1,
```
Comment (emailweixu) [inline layers/tests/test_param_sharing.py:14]: I feel the following is more natural.
self.fc1 = layers.fc(64)
self.embedding = layers.embedding((100, 64), param = self.fc1.param())

File: layers/layer_wrappers.py (hunk 1 above)
Comment (yu239-zz) [reply layers/layer_wrappers.py:136]: That might not be as flexible as now. Suppose there are two fc layers and one of them has a linear act and one of them has a sigmoid act, then the parameters cannot be shared.

Comment (yu239-zz) [reply layers/layer_wrappers.py:136]: And if one fc layer has num_flatten_dims=1 while the other has num_flatten_dims=2, then their parameters cannot be shared automatically. Unless each time we manually bind their parameters by providing one's parameters to the other's init().

File: layers/tests/test_param_name.py [hunk 3]
```diff
@@ -0,0 +1,22 @@
+import unittest
+import pprl.layers as layers
+
+
+class TestParamName(unittest.TestCase):
+    def test_name_number(self):
+        self.fc1 = layers.fc(100)
+        self.fc2 = layers.fc(100)
+        self.embedding = layers.embedding(128)
```
Comment (yu239-zz) [inline layers/tests/test_param_name.py:9]: embedding should accept a tuple or a list.

**Comment by yu239-zz** [comment]
This PR is for discussion only.

**Pull request closed without merging**